villain/parse-lib.rkt

```diff
@@ -0,0 +1,146 @@
+#lang racket
```

Comment:
Why did the parser need to be duplicated for libraries?

Reply:
When the libraries are compiled, because I have added a match clause `(LCall e es)` before `(App f es)` in parsing, this will cause an unfound label, because `compile-defines` in `compile-library` does not go through `(compile-e e c tail?)` where I have added a check for this in `compile.rkt`.  If I added a check in `parse.rkt` for checking whether `e` is `(Var f)` where `f` is a member of `stdlib-ids` this will cause a circular require, because `externs-stdlib.rkt` requires `parse.rkt`.  One solution was to add a boolean parameter to all functions in `parse.rkt` to indicate whether parsing of a library is being done.  But I just made a new file `parse-lib.rkt` where the match clause for `(LCall e es)` is commented out and the parsing of the libraries is done with this.